### PR TITLE
Ensure random Uuid generation

### PR DESCRIPTION
**Pull request by RickWinter**
Posix platforms do not generate secure random bytes with random_device().

On posix platforms use the openSSL random byte generator which generates cryptographically secure random bytes

**NOTE**: Once the assert work is done this abort() will be switched over to assert.

### CODE REVIEW

**Review by antkmsft — APPROVED** [review]

**Comment by ahsonkhan** [comment]
Once this starts passing CI, let's run the pipeline a few times for good measure. I have seen the linux randomness test fail roughly 1 in 10.

File: sdk/core/azure-core/inc/azure/core/uuid.hpp
Comment (ahsonkhan) [inline sdk/core/azure-core/inc/azure/core/uuid.hpp:20]: I believe this is introducing a dependency to the SDK we didn't previously have. Will the customer always have this header available, even if the user doesn't install `curl` from vcpkg (since it's optional)?

Do we need to update getting started?

File: sdk/core/azure-core/inc/azure/core/uuid.hpp
Comment (ahsonkhan) [inline sdk/core/azure-core/inc/azure/core/uuid.hpp:92]: This posix bug fix could benefit from a CL entry.

File: sdk/core/azure-core/inc/azure/core/uuid.hpp
Comment (RickWinter) [reply sdk/core/azure-core/inc/azure/core/uuid.hpp:20]: openssl isn't a new dependency.  It already exists for MD5.

Comment (ahsonkhan) [reply sdk/core/azure-core/inc/azure/core/uuid.hpp:20]: Ah dang it, you are right, good point.

File: sdk/core/azure-core/inc/azure/core/uuid.hpp
Comment (ahsonkhan) [inline sdk/core/azure-core/inc/azure/core/uuid.hpp:96]: Should we wait to merge this until after https://github.com/Azure/azure-sdk-for-cpp/pull/2170 so we can fix this usage?

Also, how do we plan to test this implementation?

File: sdk/core/azure-core/inc/azure/core/uuid.hpp
Comment (ahsonkhan) [inline sdk/core/azure-core/inc/azure/core/uuid.hpp:94]: nit: consider adding a comment that 0 and -1 are both considered invalid , and maybe link to: https://www.freebsd.org/cgi/man.cgi?query=RAND_bytes&sektion=3&n=1#end

**Comment by ahsonkhan** [comment]
Can you add a changelog entry for this bug fix before merging. This is an important fix that customers need to pick up.

**Pull request merged**
